src/NuGet.Clients/NuGet.CommandLine/Program.cs

```diff
@@ -231,6 +232,24 @@ private void Initialize(CoreV2.NuGet.IFileSystem fileSystem, IConsole console)
             }
         }
 
+        private Assembly CurrentDomain_ResourceResolve(object sender, ResolveEventArgs args)
+        {
+            Assembly returnedResource = null;
+
+            // We want to intercept NuGet.Resources resources and redirect it to nuget.exe assembly
```

Comment:
I'm trying to understand this change, but I need some help. 

When would the requesting assembly be NuGet.Resources? 

Is it when it's trying to resolve a string from a resources from an ilmerged assembly?

Say we're trying to resolve a resource from NuGet.Commands or NuGet.Protocol. 
It passes the first 2 conditions, but the 3rd is the one that I don't have the context for. 

I'm guessing the idea with this implementation is that if in the future we want to add say NuGet.Protocol, we'd automatically get that resolved, but just have to pay for a larger nuget.exe.

Reply:
> When would the requesting assembly be NuGet.Resources?

When the embedded `NuGet.resources.dll` resource assemblies inside nuget.exe asks for a localized resource.

https://github.com/NuGet/NuGet.Client/blob/aac22bf8d03b30c76484e65e6a37b32a45617f31/src/NuGet.Clients/NuGet.CommandLine/NuGet.CommandLine.csproj#L83

> Is it when it's trying to resolve a string from a resources from an ilmerged assembly?

Citing ILMerge docs, when you're _ilmerging_ assemblies, resources are just copied to the final ilmerge'd assembly.

https://github.com/dotnet/ILMerge/blob/9ec1b3f29f6f11b8fc2ddcca4c055dae7b10d9e3/ilmerge-manual.md?plain=1#L18

Now, since the resource lookup starts in NuGet.resources.dll embedded resource assembly; this assembly starts looking for resources and triggers `CurrrentDomain_ResourceResolve` event. 

1. If the resource is stored inside `NuGet.resources.dll`, we don't want to intercept, we just the resource lookup do its job (first condition). Condition is: `!args.Name.StartsWith("NuGet.CommandLine", StringComparison.OrdinalIgnoreCase)`
2. But, if we see that the requested resource name starts with "NuGet", then, we are interested in loading from the ilmerged resources (second condition).  Condition is: `args.Name.StartsWith("NuGet", StringComparison.OrdinalIgnoreCase)`
3. And, we want to intercept the event only if resource lookups only coming from NuGet.resources.dll resource assembly (third condition). If the event comes from another assembly, we don't have enough information to make a decision and let the resource lookup algorithm do its job. Condition is: `string.Equals("NuGet.Resources", args.RequestingAssembly.GetName().Name, StringComparison.OrdinalIgnoreCase)`

### ILSpy with example

![ilspy-with-notes](https://user-images.githubusercontent.com/1192347/204951649-c9e381fe-f13a-441a-9a99-d2749002840d.png)

Reply:
> If the event comes from another assembly, we don't have enough information to make a decision and let the resource lookup algorithm do its job

So when a string from Nuget.Commands originally is getting looked up, I'm guessing NuGet.resources will always be the requesting assembly since it comes from within the ilmerged NUGet.exe?

If yes, then I think I get it. :D

Reply:
Yes, that's what I observed during debugging.

For more info, refer to: https://learn.microsoft.com/en-us/dotnet/core/extensions/package-and-deploy-resources#net-framework-resource-fallback-process